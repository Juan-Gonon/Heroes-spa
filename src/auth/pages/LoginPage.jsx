import { useNavigate } from 'react-router-dom'
import { useAuth } from '../../hooks/useAuth'

export function LoginPage () {
  const navigate = useNavigate()
  const { login } = useAuth()

  const handleClickLogin = () => {
    const lastPath = localStorage.getItem('lastPath') || '/'
    login('Juan')
    navigate(lastPath, {
      replace: true
    })
  }

  return (
    <div className='container mt-5'>
      <h1>Login</h1>
      <hr />
      <button className='btn btn-primary' onClick={handleClickLogin}>Login</button>
    </div>
  )
}
