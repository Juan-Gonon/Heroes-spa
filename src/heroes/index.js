// eslint-disable-next-line import/export
export * from './pages'
export * from './routes/HeroesRoutes'
