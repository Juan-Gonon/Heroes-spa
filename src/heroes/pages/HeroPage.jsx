import { Navigate, useNavigate, useParams } from 'react-router-dom'
import { getHeroById } from '../helpers'
import { useMemo } from 'react'

export function HeroPage () {
  const { id } = useParams()
  const navigate = useNavigate()
  const hero = useMemo(() => {
    return getHeroById({ id })
  }, [id])

  if (!hero) return <Navigate to='/marvel' />

  const handleClickBack = () => {
    navigate(-1)
  }

  console.log(hero)
  return (
    <div className='row mt-5 '>
      <div className='col-4'>
        <img className='img-thumbnail animate__animated animate__fadeInLeft' src={`/src/assets/heroes/${id}.jpg`} alt='' />
      </div>

      <div className='col-8'>
        <h3>{hero.superhero}</h3>
        <ul className='list-group list-group-flush'>
          <li className='list-group-item'>
            <b>Alter ego:</b> {hero.alter_ego}
          </li>
          <li className='list-group-item'>
            <b>Publisher:</b> {hero.publisher}
          </li>
          <li className='list-group-item'>
            <b>First Appearance:</b> {hero.first_appearance}
          </li>
        </ul>
        <div className='mt-3'>Characters</div>
        <p>{hero.characters}</p>
        <button className='btn btn-outline-primary' onClick={handleClickBack}>Regresar</button>
      </div>

    </div>
  )
}
