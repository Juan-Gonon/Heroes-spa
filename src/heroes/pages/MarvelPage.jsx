import { HeroList } from '../components/HeroList'
export function MarvelPage () {
  return (
    <>
      <h1>Marvel Comics</h1>
      <hr />
      <HeroList publisher='Marvel Comics' />
    </>
  )
}
