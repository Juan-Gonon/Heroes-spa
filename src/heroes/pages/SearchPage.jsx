import { HeroCard } from '../components'
import { useLocation, useNavigate } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import queryString from 'query-string'
import { getHeroesByName } from '../helpers'

export function SearchPage () {
  const navigate = useNavigate()
  const location = useLocation()

  const { q = '' } = queryString.parse(location.search)
  const { searchText, onInputChange } = useForm({
    searchText: q
  })

  const heroes = getHeroesByName({ name: q })
  // console.log(heroes)

  const handleSubmitForm = (e) => {
    e.preventDefault()
    // if (searchText.trim().length < 1) return
    // console.log(searchText)
    navigate(`?q=${searchText}`)
  }

  const showSearch = (q.length === 0)
  const showError = (q.length > 0) && heroes.length === 0

  return (
    <>
      <h1>Search</h1>
      <hr />
      <div className='row'>

        <div className='col-5'>
          <h4>Searching</h4>
          <hr />
          <form aria-label='form-event' onSubmit={handleSubmitForm}>
            <input
              type='text'
              placeholder='Search a hero'
              className='form-control'
              name='searchText'
              autoComplete='off'
              value={searchText}
              onChange={onInputChange}
              aria-label='searchText'
            />
            <button className='btn btn-outline-primary mt-1 '>Search</button>
          </form>
        </div>
        <div className='col-7'>
          <h4>Results</h4>
          <hr />
          {/* {
            (q === '')
              ? (
                <div className='alert alert-primary '>
                  Search a hero
                </div>
                )
              : (heroes.length === 0) && (
                <div className='alert alert-danger'>
                  No hero With <b>{q}</b>
                </div>
                )
          } */}

          <div
            aria-label='msg-search-hero'
            className='alert alert-primary animate__animated animate__fadeIn ' style={{
              display: showSearch ? '' : 'none'
            }}
          >
            Search a hero
          </div>

          <div
            aria-label='alert-not-hero'
            className='alert alert-danger animate__animated animate__fadeIn' style={{
              display: showError ? '' : 'none'
            }}
          >
            No hero With <b>{q}</b>
          </div>

          {
            heroes.map((hero) => (
              <HeroCard key={hero.id} {...hero} />

            ))
          }

        </div>
      </div>
    </>
  )
}
