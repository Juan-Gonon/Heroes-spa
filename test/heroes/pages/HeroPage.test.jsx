import { afterEach, describe, expect, it, vi } from 'vitest'
import { cleanup, render, screen } from '@testing-library/react'
import { HeroPage } from '../../../src/heroes/pages'
import { useParams } from 'react-router-dom'

vi.mock('react-router-dom')
// const useParamsMock = vi.fn().mockReturnValue({ id: 'marvel-spider' })

describe('Prueba en <HeroPage /> ', () => {
  afterEach(() => {
    cleanup()
  })
  it('debe de retornar el hero spider ', () => {
    useParams.mockReturnValue({
      id: 'marvel-spider'
    })
    const { container } = render(
      <HeroPage />
    )

    expect(container).toMatchSnapshot()
  })

  it('debe de mostrar la imagen de spider', () => {
    useParams.mockReturnValue({
      id: 'marvel-spider'
    })

    render(
      <HeroPage />
    )

    const img = screen.getByLabelText('img-hero')

    expect(img.src).toContain('/src/assets/heroes/marvel-spider.jpg')
  })

  // it('debe de llamar el navigate', () => {
  //   useParams.mockReturnValue({
  //     id: 'marvel-spider'
  //   })

  //   render(
  //     <HeroPage />
  //   )

  //   const btnPrev = screen.getByRole('button')
  //   fireEvent.click()
  // })
})
