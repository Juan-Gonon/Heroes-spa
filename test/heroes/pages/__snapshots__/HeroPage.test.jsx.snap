// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Prueba en <HeroPage />  > debe de retornar el hero spider  1`] = `
<div>
  <div
    class="row mt-5 "
  >
    <div
      class="col-4"
    >
      <img
        alt=""
        aria-label="img-hero"
        class="img-thumbnail animate__animated animate__fadeInLeft"
        src="/src/assets/heroes/marvel-spider.jpg"
      />
    </div>
    <div
      class="col-8"
    >
      <h3>
        Spider Man
      </h3>
      <ul
        class="list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
          <b>
            Alter ego:
          </b>
           
          Peter Parker
        </li>
        <li
          class="list-group-item"
        >
          <b>
            Publisher:
          </b>
           
          Marvel Comics
        </li>
        <li
          class="list-group-item"
        >
          <b>
            First Appearance:
          </b>
           
          Amazing Fantasy #15
        </li>
      </ul>
      <div
        class="mt-3"
      >
        Characters
      </div>
      <p>
        Peter Parker
      </p>
      <button
        class="btn btn-outline-primary"
      >
        Regresar
      </button>
    </div>
  </div>
</div>
`;

exports[`Prueba en <HeroPage />  > debe de retornar el hero spider 1`] = `<div />`;
